apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -53,21 +57,28 @@
 
   private static final Logger log = LoggerFactory.getLogger(ApiarySnsListener.class);
 
+  static final String PROTOCOL_VERSION = "1.0";
+
   private static final String TOPIC_ARN = System.getenv("SNS_ARN");
-  final static String PROTOCOL_VERSION = "1.0";
-  private final String protocolVersion = PROTOCOL_VERSION;
+
+  private final String tableParamFilter = System.getenv("TABLE_PARAM_FILTER");
+  private final Pattern tableParamFilterPattern;
 
   private final AmazonSNS snsClient;
 
   public ApiarySnsListener(Configuration config) {
-    super(config);
-    snsClient = AmazonSNSClientBuilder.defaultClient();
-    log.debug("ApiarySnsListener created ");
+    this(config, AmazonSNSClientBuilder.defaultClient());
   }
 
   ApiarySnsListener(Configuration config, AmazonSNS snsClient) {
     super(config);
     this.snsClient = snsClient;
+    tableParamFilterPattern = (tableParamFilter == null) ? null : Pattern.compile(tableParamFilter);
+
+    if (tableParamFilterPattern != null) {
+      log.info(String.format("Environment Variable TABLE_PARAM_FILTER is set as [%s]", tableParamFilter));
```

Comment:
You might as well change the if statement to check `tableParamFilter` for null, then you can move the setting and compile of the pattern from line 76 into here and then you won't need to use the ternary operator.

Reply:
`tableParamFilterPattern` would no longer be final but I'm OK with that ;)

Reply:
Very well spotted :)

Reply:
Glad my old hawk eyes still work!

Reply:
🦅